### ISSUE TITLE
Episode sorting / sequence issue: Addition of a leading zero for triple plus digit episode numbers?

### ISSUE THREAD

**Issue by demosthien** [Komet/MediaElch#610]
I've just noticed that for some TV Series with some episodes (in the triple digits) there is no leading zero added to the filename. For example, the Doctor Who (2005) specials now number into 154 or so episodes. So, the most recent Christmas special has a filename on my system as follows:

`Doctor Who (2005) S00E154 - Twice Upon A Time.mp4`

It seems that Kodi has been hiding these from me, my Doctor Who Specials only appeared to reach episode 99, as it has them all displayed non-sequentially (well probably sequential in some weird computer way so they were way up the list near the middle). Anyway, I assume this is something to do with Mediaelch not adding a leading zero before the 154 here - `S00E154`

Is that correct? Or is it something I should be chasing Kodi support about?

**Comment by demosthien** [comment]
Cheers!

Yes, something like that. I've not got all the files in sequence so some are missing and it looks a bit weird but here's how it goes so you get an idea;

`Doctor Who (2005) S00E09 - The Next Doctor
Doctor Who (2005) S00E104 - The Doctors Revisited The First Doctor
Doctor Who (2005) S00E105 - The Doctors Revisited The Second Doctor
Doctor Who (2005) S00E107 - The Doctors Revisited The Fourth Doctor
Doctor Who (2005) S00E114 - The Doctors Revisited The Eleventh Doctor
Doctor Who (2005) S00E13 - Planet of the Dead
Doctor Who (2005) S00E14 - The Waters of Mars
Doctor Who (2005) S00E142 - Last Christmas
Doctor Who (2005) S00E149 - The Return of Doctor Mysterio
Doctor Who (2005) S00E15 - Dreamland
Doctor Who (2005) S00E154 - Twice Upon A Time
Doctor Who (2005) S00E16 - The End of Time (1)`

**Comment by bugwelle** [comment]
Yep, got it. This is a desirable feature :-)
Can't say when I'll have time for this, though.

**Comment by demosthien** [comment]
No problem. I can’t see it being a serious issue requiring quick action given it only really affects series’ episodes reaching into the 100s, and there won’t be too many of those around.

**Comment by bugwelle** [comment]
No, each episode-number has to have the same number of digits (including leading zeros) to make sorting work. :-)
